Backlog clearing: 8 months of SC updates! 2023-11 through 2024-06-19 (#249)
python/steering-council: updates/2024-04,05,06-steering-council-update.md

# April through June 19, 2024 Steering Council update

- Steering Council discussion topics:
    - **PEPs**
        - Ongoing [PEP 741, a Configuration C API](https://peps.python.org/pep-0741/) discussions; seeking consensus and clarity and WG input. Ultimately decided to postpone this until 3.14.
        - Repeated check-ins on [PEP 649, deferred evaluation of annotations using descriptors](https://peps.python.org/pep-0649/) implementation. We're collectively okay if this slips into 3.14.
        - Finalized and posted our decision on [PEP 734, subinterpreters in the stdlib](https://discuss.python.org/t/pep-734-multiple-interpreters-in-the-stdlib/41147/24).
        - Accepted [PEP 667, Consistent views of namespaces](https://peps.python.org/pep-0667/) after much discussion. The alternatives were considered worse in the long run.
    - Sync up meetings with Łukasz, the Developer-in-Residence:

>         - The untangling of potential license and CLA issues on the infrastructure repositories. Not glamourous work, needs doing, will involve PSF council as appropriate.

Reviewer: ```suggestion
        - The untangling of potential license and CLA issues on the infrastructure repositories. Not glamorous work, needs doing, will involve PSF counsel as appropriate.
```

    - Discussions around our Communications Liaison hiring. PSF working out the contract. Our initial offer fell through, we [extended an offer to our alternate equally qualified candidate](https://discuss.python.org/t/the-steering-council-is-hiring/44686/6), as we had discussed might happen with them during our initial rejection communication. The title was renamed from Secretary to Communications Liaison to better reflect the role.
    - Discussed how it'd be nice if our monthly SC update summaries were actually created, reviewed, and published before PyCon US 2024. *(editors note: LOL... given the backlog this is being posted as part of, volunteer time is precious, thus hiring)*
    - Discussed new 3.13 REPL platform implementation woes. "At least it is not using readline" also means "people who rely on obscure readline featureitis frown".
    - Discussions of type annotations in the stdlib. More community discussions needed before any proposal can be made. Annotations in `Tools/` have found actual bugs.
    - Discussed capacity issues at the upcoming September core dev sprint; space limitations are real and lead to hard choices.

>     - Discussed logistics of the remaining two core dev. mentorship sessions that Edward is holding this quarter. *(editors note: held)*

Reviewer: ```suggestion
    - Discussed logistics of the remaining two core dev mentorship sessions that Edward is holding this quarter. *(editors note: held)*
```

    - Discussed further ongoing core dev mentorship needs and the utility of a possible Mentorship WG. One theme has always been lack of volunteer time availability.

>     - Discussed how it'd be good if release managers wrote up retrospectives. That requires extra time committment on their part. We'd all benefit, not just future RMs.

Reviewer: ```suggestion
    - Discussed how it'd be good if release managers wrote up retrospectives. That requires extra time commitment on their part. We'd all benefit, not just future RMs.
```

    - Discussed our recent core dev promotion votes. No vetoes (all approved). We're noticing that our process feels a bit odd, the SC represents the community thus a veto of any form would not be well received. Nor would a vote itself failing. So this means anyone proposing someone effectively needs to do homework and talk via backchannels to a slew of core devs and SC first to ensure that neither negative outcome is likely. Maybe this is working as intended? But we don't state it as such anywhere... But what would be meaningfully better to propose?
    - Discussed and announced who would be our next Release Manager (RM) and related logistical issues.
    - Continual brainstorming on what to include in what form for our PyCon US 2024 keynote address. In classic PyCon fashion our collective presentation was congealing up until the day we gave it.
    - Discussed a PR brought to our attention that we believe should not have been merged as is, regardless of intent, and whether or not we can do anything about it right now.

>     - More discussions community friction regarding Packaging and PyPI, the Packaging WG, and PyPA with the PSF.

Reviewer: ```suggestion
    - More discussions on community friction regarding Packaging and PyPI, the Packaging WG, and PyPA with the PSF.
```

    - Noticing how we have misaligned cadences among many of our collective core dev working styles. It leads to a strain on decision making. Perhaps a discussion topic for the core dev sprint at Meta?

>     - Greg and Barry employer changes; no [PEP-13 conflicts](https://peps.python.org/pep-0013/#conflicts-of-interest).

Reviewer: ```suggestion
    - Greg and Barry employer changes; no [PEP 13 conflicts](https://peps.python.org/pep-0013/#conflicts-of-interest).
```
